### PR TITLE
[AGENT-4241] Implement Azure Credentials management

### PR DESCRIPTION
**Pull request by oleksandr-saienko**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Added support for Azure Storage Credentials Management from Airflow

## Rationale
Users should be able to configure and use DataRobot Azure Storage Credentials from Airflow UI

### CODE REVIEW

**Comment by oleksandr-saienko** [comment]
Jenkins run

**Comment by oleksandr-saienko** [comment]
Jenkins run

File: datarobot_provider/hooks/credentials.py [hunk 1]
```diff
@@ -328,3 +328,78 @@ def get_ui_field_behaviour() -> Dict:
...
+
+        if not conn.login:
+            raise AirflowException("Storage Account Name is not defined")
+
+        if not conn.password:
+            raise AirflowException("Storage Account Key is not defined")
+
+        try:
+            self.log.info(
+                f"Creating Azure Storage Credentials:{self.datarobot_credentials_conn_id}"
```
Comment (elatt) [inline datarobot_provider/hooks/credentials.py:348]:
```suggestion
                f"Creating Azure Storage Credentials: {self.datarobot_credentials_conn_id}"
```
nit

File: datarobot_provider/hooks/credentials.py [hunk 2]
```diff
@@ -328,3 +328,78 @@ def get_ui_field_behaviour() -> Dict:
...
+                    conn.login, conn.password
+                ),
+                description=self.default_credential_description,
+            )
+
+            return credential
+
+        except Exception as e:
+            self.log.error(
+                f"Error creating Azure Credentials: {self.datarobot_credentials_conn_id}, message:{str(e)}"
```
Comment (elatt) [inline datarobot_provider/hooks/credentials.py:363]:
```suggestion
                f"Error creating Azure Credentials: {self.datarobot_credentials_conn_id}, message:{e}"
```
Another minor nit

**Review by elatt — APPROVED** [review]

**Review by dulcardo — APPROVED** [review]

**Pull request merged**